core/sql/executor/HBaseClient_JNI.cpp

```diff
@@ -2844,819 +2871,980 @@ HBC_RetCode HBaseClient_JNI::getHbaseTableInfo(const char* tblName,
- ,"Java exception in getRowsOpen()."
- ,"Java exception in getClose()."
- ,"Preparing parameters for deleteRow()."
- ,"Java exception in deleteRow()."
- ,"Preparing parameters for deleteRows()."
- ,"Java exception in deleteRows()."
- ,"Preparing parameters for checkAndDeleteRow()."
- ,"Java exception in checkAndDeleteRow()."
- ,"Row deleted in checkAndDeleteRow()."
- ,"Preparing parameters for insertRow()."
- ,"Java exception in insertRow()."
- ,"Preparing parameters for insertRows()."
- ,"Java exception in insertRows()."
- ,"Preparing parameters for checkAndInsertRow()."
- ,"Java exception in checkAndInsertRow()."
- ,"Dup RowId in checkAndInsertRow()."
- ,"Transactions not supported yet in checkAndInsertRow()"
- ,"Preparing parameters for checkAndUpdateRow()."
- ,"Java exception in checkAndUpdateRow()."
- ,"Dup RowId in checkAndUpdateRow()."
- ,"Transactions not supported yet in checkAndUpdateRow()"
- ,"Preparing parameters for create()."
- ,"Java exception in create()."
- ,"Preparing parameters for drop()."
- ,"Java exception in drop()."
- ,"Preparing parameters for exists()."
- ,"Java exception in exists()."
- ,"Preparing parameters for coProcAggr()."
- ,"Java exception in coProcAggr()."
- ,"Preparing parameters for grant()."
- ,"Java exception in grant()."
- ,"Preparing parameters for revoke()."
- ,"Java exception in revoke()."
- ,"Java exception in getendkeys()."
- ,"Java exception in getHTableName()."
- ,"Java exception in flush()."
- ,"Java exception in getColName()."
- ,"Java exception in getColValue()."
- ,"Java exception in getRowID()."
- ,"Java exception in nextCell()."
- ,"Preparing parameters for getRows()."
- ,"Java exception in getRows()."
- ,"Java exception in completeAsyncOperation()."
- ,"Async Hbase Operation not yet complete."
-};
-
 //////////////////////////////////////////////////////////////////////////////
 // 
 //////////////////////////////////////////////////////////////////////////////
-char* HTableClient_JNI::getErrorText(HTC_RetCode errEnum)
+HBC_RetCode HBaseClient_JNI::insertRow(NAHeap *heap, const char *tableName,
+      ExHbaseAccessStats *hbs, bool useTRex, Int64 transID, HbaseStr rowID,
+      HbaseStr row, Int64 timestamp, bool checkAndPut, bool asyncOperation,
+      HTableClient_JNI **outHtc)
 {
-  if (errEnum < (HTC_RetCode)JOI_LAST)
-    return JavaObjectInterface::getErrorText((JOI_RetCode)errEnum);
-  else    
-    return (char*)htcErrorEnumStr[errEnum-HTC_FIRST-1];
-}
+  
+  HTableClient_JNI *htc = NULL;
 
-//////////////////////////////////////////////////////////////////////////////
-// 
-//////////////////////////////////////////////////////////////////////////////
-HTableClient_JNI::~HTableClient_JNI()
-{
-  //QRLogger::log(CAT_JNI_TOP, LL_DEBUG, "HTableClient destructor called.");
-  cleanupResultInfo();
-  if (tableName_ != NULL)
-  {
-     NADELETEBASIC(tableName_, heap_);
+  if (asyncOperation) {
+     HTableClient_JNI *htc = new (heap) HTableClient_JNI(heap, (jobject)-1);
```

Comment:
Should this just be htc = ... ?
In line 2883 we have already declared the htc variable.